router/sleeve.go

```diff
@@ -287,9 +291,12 @@ type sleeveForwarder struct {
 	confirmedChan    chan<- struct{}
 	finishedChan     <-chan struct{}
 
+	// listener channels
+	establishedChan chan struct{}
+	errorChan       chan error
```

Comment:
D'oh, we need to write to them of course, but return them as read only to the API consumer...